### PR TITLE
3538 - 0

### PR DESCRIPTION
**Pull request by KAMEK0NA**
## Номер
3538

## Номер задания
0

## Ссылка на видео с демо
файл в папке

## Комментарии
Хотел сделать то-то, но не успел и не понимаю.

### CODE REVIEW

File: 3538/0/triangle.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#   about_triangle_project.rb
+# and
+#   about_triangle_project_2.rb
+#
+def triangle(side_a, side_b, side_c)
+  # :reek:TooManyStatements
+  # :reek:FeatureEnvy
+  side_a, side_b, side_c = [side_a, side_b, side_c].sort
+  raise TriangleError unless side_a.positive?
+  raise TriangleError unless side_a + side_b > side_c
```
Comment (DmitryAvramec) [inline 3538/0/triangle.rb]: Better use `if`

File: 3538/0/triangle.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  return :equilateral if side_a == side_c
+  return :isosceles if side_a == side_b || side_b == side_c
```
Comment (DmitryAvramec) [inline 3538/0/triangle.rb]: You have Array, use it

File: 3538/0/about_scoring_project.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+# score([1,5,1,2,4]) => 250 points
+#
+# More scoring examples are given in the tests below:
+#
+# Your goal is to write the score method.
+
+def score(dice)
+  dice = dice.sort
+  sum = 0
+  (1..6).each do |round|
```
Comment (DmitryAvramec) [inline 3538/0/about_scoring_project.rb]: Reduce, inject or each_with_object

File: 3538/0/about_scoring_project.rb [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    count = dice.count(round)
+    sum += (round == 1 ? 1000 : round * 100) if count >= 3
+    sum += (count % 3) * 100 if round == 1
```
Comment (DmitryAvramec) [inline 3538/0/about_scoring_project.rb]: Use ternary operator, 2 row is complex here. And you can find a better solution for this logic

File: 3538/0/triangle.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+# The tests for this method can be found in
+#   about_triangle_project.rb
+# and
+#   about_triangle_project_2.rb
+#
+def triangle(side_a, side_b, side_c)
+  # :reek:TooManyStatements
+  # :reek:FeatureEnvy
+  sides = [side_a, side_b, side_c].sort
+  raise TriangleError if sides[0] <= 0
```
Comment (DmitryAvramec) [inline 3538/0/triangle.rb]: O_o 0, 1, 2
Go away from digits

File: 3538/0/.rubocop.yml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+Style/FrozenStringLiteralComment:
```
Comment (DmitryAvramec) [inline 3538/0/.rubocop.yml]: Why this file can ever be here?

File: 3538/0/about_scoring_project.rb [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+# score([3,4,5,3,3]) => 350 points
+# score([1,5,1,2,4]) => 250 points
+#
+# More scoring examples are given in the tests below:
+#
+# Your goal is to write the score method.
+
+def score(dice)
+  sum = 0
+  sum += dice.count(1) >= 3 ? 1000 + (dice.count(1) - 3) * 100 : dice.count(1) * 100
```
Comment (DmitryAvramec) [inline 3538/0/about_scoring_project.rb]: Why you move this logic away from iterator? You can easily make it work inside of `(1..6).each_with_object(0) do |number, score|` and you don't even need `sum = 0` and other initialization of it before you start iterating on numbers through dices.

**Comment by KAMEK0NA** [comment]
All comments saved and will be taken into account at the next PR

**Pull request closed without merging**